Style Guide fixes, typos, and fixed links (#4437)
braze-inc/braze-docs: _docs/_developer_guide/platform_wide/sdk_authentication.md

| Field | Required | Description                                                                            |
| ----- | -------- | -------------------------------------------------------------------------------------- |

> | `sub` | **Yes**  | The "subject" should equal the User ID you supply Braze SDK when calling `changeUser`  |
> | `exp` | **Yes**  | The "expiration" of when you want this token to expire.                                |

Reviewer: ```suggestion
| `sub` | Yes  | The "subject" should equal the User ID you supply Braze SDK when calling `changeUser`  |
| `exp` | Yes | The "expiration" of when you want this token to expire.                                |
```

### Set the current user's JWT token

Whenever your app calls the Braze [`changeUser`][11] method, also supply the JWT token that was [generated server-side][4].

You can also configure the token to refresh mid-session for the current user.

{% alert note %}
Keep in mind that [`changeUser`][11] should only be called when the User ID has _actually changed_. You should not use this method as a way to update the signature if the user ID has not changed.
{% endalert %}

{% tabs %}
{% tab Javascript %}
Supply the JWT Token when calling [`changeUser`][11]:

```javascript

> import * as braze from"@braze/web-sdk";

Reviewer 2: ```suggestion
import * as braze from "@braze/web-sdk";
```

When this feature is set as ["Required"](#enforcement-options), the following scenarios will cause SDK requests to be rejected by Braze:
- JWT was expired by the time is was received by the Braze API
- JWT was empty or missing
- JWT failed to verify for the public keys you uploaded to the Braze dashboard

You can use [`subscribeToSdkAuthenticationFailures`](https://js.appboycdn.com/web-sdk/latest/doc/modules/braze.html#subscribetosdkauthenticationfailures) to subscribe to be notified when the SDK requests fail for one of these reasons. A callback function contains an object with the relevant [`errorCode`][9], `reason` for the error, the `userId` of the request (if the user is not anonymous), and the authentication `signature` that caused the error. 

> Failed requests will periodically be retried until your app supplies a new valid JWT. If _that user_ is still logged in, you can use this callback as an opportunity to request a new JWT from your server and supply Braze's SDK with this new valid token.

Reviewer: ```suggestion
Failed requests will periodically be retried until your app supplies a new valid JWT. If that user is still logged in, you can use this callback as an opportunity to request a new JWT from your server and supply Braze's SDK with this new valid token.
```

{% tab Javascript %}
```javascript
import * as braze from"@braze/web-sdk";

> braze.subscribeToSdkAuthenticationFailures((someErrorEvent) => {

Reviewer 2: ```suggestion
braze.subscribeToSdkAuthenticationFailures((errorEvent) => {
```

Reviewer 2: these 3 suggestions should be applied to other code samples below.

  // TODO: Optionally log to your error-reporting service

>   // TODO: Check if the someErrorEvent user matches the currently logged-in user

Reviewer 2: ```suggestion
  // TODO: Check if the user_id within the `errorEvent` matches the currently logged-in user
```

>   const updated_jwt = await getNewTokenSomehow(someErrorEvent);

Reviewer 2: ```suggestion
  const updated_jwt = await getNewTokenSomehow(errorEvent);
```

```dart
braze.setBrazeSdkAuthenticationErrorCallback((BrazeSdkAuthenticationError error) async {
  // TODO: Optionally log to your error-reporting service
  // TODO: Check if the `user_id` within the `error` matches the currently logged-in user
  print("Invalid SDK Authentication signature.")
  let newSignature = getNewSignatureSomehow(error)
  braze.setSdkAuthenticationSignature(newSignature);
});
```
{% endtab %}
{% endtabs %}

## Adding public keys {#key-management}

> In the **Manage Settings** page of the dashboard, add your public key to a specific app in the Braze dashboard. Each app supports up to three public keys. Note that the same public/private keys may be used across apps.

Reviewer: ```suggestion
In the **Manage Settings** page of the dashboard, add your public key to a specific app in the Braze dashboard. Each app supports up to three public keys. Note that the same public and private keys may be used across apps.
```

To add a public key:

1. Choose the app from the list of available apps.
2. Under **SDK Authentication**, click **Add Public Key**.
3. Paste in the public key, and add an optional description.
4. After saving your changes, the key will appear in the list of public keys.

To delete a key, or to promote a key to the primary key, choose the corresponding action in the overflow menu next to each key.

## Enabling in the Braze dashboard {#braze-dashboard}

Once your [Server-side Integration][1] and [SDK Integration][2] are complete, you can begin to enable this feature for those specific apps.

> Keep in mind, SDK requests will continue to flow as usual - without authentication - _unless_ the app's SDK Authentication setting is set to **Required** in the Braze dashboard.

Reviewer: ```suggestion
Keep in mind that SDK requests will continue to flow as usual without authentication unless the app's SDK Authentication setting is set to **Required** in the Braze dashboard.
```

> Should anything go wrong with your integration (i.e., your app is incorrectly passing tokens to the SDK, or your server is generating invalid tokens), simply disable this feature in the Braze dashboard and data will resume to flow as usual, without verification.

Reviewer: ```suggestion
Should anything go wrong with your integration (i.e., your app is incorrectly passing tokens to the SDK, or your server is generating invalid tokens), disable this feature in the Braze dashboard, and data will resume to flow as usual without verification.
```
